RomanNumberConverter/src/romannumberconverter/RomanNumberConverter.java

```diff
@@ -0,0 +1,144 @@
+package romannumberconverter;
+
+/**
+ *
+ * @author Santiago
+ */
+public class RomanNumberConverter {
+
+    private static String romanNumber;
+
+    public enum singleChar {
+
+        I(1),
+        V(5),
+        X(10),
+        L(50),
+        C(100),
+        D(500),
+        M(1000);
+
+        int value;
+
+        int getValue() {
+            return value;
+        }
+
+        singleChar(int value) {
+            this.value = value;
+        }
+
+        public static singleChar valueOf(char romanChar) {
+            switch (String.valueOf(romanChar).toLowerCase().toCharArray()[0]) {
```

Comment:
Tendriamos que poner la enum afuera en una archivo separado y llamarla RomanNumber. Luego en ves de tener un switch se podria tener un bucle while que vaya comprando e iterando y corte cuando encuentra el elemento. Existe un metodo que se llama enumClass.values()

public static RomanNumber getRomanValueFromChar(Char i) {
     for (RomanNumber romanNumber : RomanNumber.values()) {
         if (romanNumber.getValue().equals(i)) {
             return status.toString();
         }
     }
     // throw an IllegalArgumentException or return null
     throw new IllegalArgumentException("the given letter doesn't match any Roman Number.");
 }